src/main/java/io/debezium/connector/yugabytedb/YugabyteDBTablePoller.java

```diff
@@ -0,0 +1,138 @@
+ * server side to the provided stream ID.
+ * 
+ * @author Vaibhav Kushwaha (vkushwaha@yugabyte.com)
+ */
+public class YugabyteDBTablePoller extends Thread {
+  private static final Logger LOGGER = LoggerFactory.getLogger(YugabyteDBTablePoller.class);
+  private final short MAX_RETRY_COUNT = 5;
+
+  private final YugabyteDBConnectorConfig connectorConfig;
+  private final ConnectorContext connectorContext;
+  private final YBClient ybClient;
+  private final CountDownLatch shutdownLatch;
+  private final long pollMs;
+
+  private short retryCount = 0;
+  
+  private List<TableInfo> cachedTableInfoList = null;
+
+  public YugabyteDBTablePoller(YugabyteDBConnectorConfig connectorConfig,
+                               ConnectorContext connectorContext) {
+    super();
+    this.connectorConfig = connectorConfig;
+    this.connectorContext = connectorContext;
+    this.ybClient = YBClientUtils.getYbClient(connectorConfig);
+    this.shutdownLatch = new CountDownLatch(1);
+    this.pollMs = connectorConfig.newTablePollIntervalMs();
+  }
+
+  @Override
+  public void run() {
+    LOGGER.info("Starting thread to monitor the tables");
+    while (shutdownLatch.getCount() > 0) {
+      if (areThereNewTablesInStream()) {
+        this.connectorContext.requestTaskReconfiguration();
+      }
+
+      try {
+        LOGGER.debug("Waiting for {} ms to poll again for new tables", pollMs);
+        boolean shuttingDown = shutdownLatch.await(pollMs, TimeUnit.MILLISECONDS);
+        if (shuttingDown) {
+          return;
+        }
+      } catch (InterruptedException ie) {
+        LOGGER.error("Unexpected interrupted exception, ignoring", ie);
+      }
+    }
+  }
+
+  /**
+   * Verify if there are any new tables added to the change data stream. This function continuously
+   * fetches the DB stream info for a given stream and as soon as there is a change in the stream
+   * info, it returns a signal to the connector.
+   * @return true if there is a new table in the stream info
+   */
+  private boolean areThereNewTablesInStream() {
+    try {
+      GetDBStreamInfoResponse resp = this.ybClient.getDBStreamInfo(this.connectorConfig.streamId());
+
+      // Reset the retry counter.
+      retryCount = 0;
+
+      if (cachedTableInfoList == null) {
+        LOGGER.debug("Cached table list in the poller thread is null, initializing it now");
+        cachedTableInfoList = resp.getTableInfoList();
+      } else {
+        if (cachedTableInfoList.size() != resp.getTableInfoList().size()) {
+          // TODO: We can also check for a condition whether the new added tables are a part of the
+          // include list (if they satisfy the regex criteria), if they don't satisfy then we
+          // should not restart.
+
+          String message = "Found {} new table(s), signalling context reconfiguration";
+          LOGGER.info(message, resp.getTableInfoList().size() - cachedTableInfoList.size());
```

Comment:
If you decide to use Set instead of List, can you print intersection & difference if debug logging is turned on? 
Refer: https://stackoverflow.com/a/3590718

Reply:
Added the code to print intersection and difference while debug logging is on.